tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -0,0 +1,91 @@
+/// \file RNTupleProcessor.cxx
+/// \ingroup NTuple ROOT7
+/// \author Florine de Geus <florine.de.geus@cern.ch>
+/// \date 2024-03-26
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2024, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <ROOT/RNTupleProcessor.hxx>
+
+#include <ROOT/RField.hxx>
+
+ROOT::Experimental::Internal::RNTupleProcessorContext::RNTupleProcessorContext(const RNTupleSourceSpec &ntuple)
+{
+   fPageSource = RPageSource::Create(ntuple.fName, ntuple.fStorage);
+   fPageSource->Attach();
+   fModel = fPageSource->GetSharedDescriptorGuard()->CreateModel();
+   fEntry = fModel->CreateBareEntry();
+}
+
+void ROOT::Experimental::Internal::RNTupleProcessorContext::Initialize()
+{
+   fPageSource->Attach();
+   auto desc = fPageSource->GetSharedDescriptorGuard();
+
+   auto fnSetProcessorFields = [&](DescriptorId_t fieldId, auto &setFunc) -> void {
+      auto &fieldDesc = desc->GetFieldDescriptor(fieldId);
+
+      auto fieldOrException = RFieldBase::Create(fieldDesc.GetFieldName(), fieldDesc.GetTypeName());
+      if (fieldOrException) {
+         auto field = fieldOrException.Unwrap();
+         field->SetOnDiskId(fieldId);
+         fFieldContexts.emplace_back(std::move(field), fEntry->GetToken(fieldDesc.GetFieldName()));
+      }
+
+      for (const auto &subFieldDesc : desc->GetFieldIterable(fieldId)) {
+         setFunc(subFieldDesc.GetId(), setFunc);
+      }
```

Comment:
Not quite sure about the recursion: `RFieldBase::Create` already creates the sub fields.